nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-site-to-site/src/main/java/org/apache/nifi/remote/protocol/AbstractFlowFileServerProtocol.java

```diff
@@ -375,7 +375,7 @@ protected int commitTransferTransaction(Peer peer, FlowFileTransaction transacti
             throw new ProtocolException("After sending data, expected TRANSACTION_FINISHED response but got " + transactionResponse);
         }
 
-        session.commit();
+        session.commitAsync();
```

Comment:
Yeah that was intentionally left as `session.commit()`. After committing the session, it responds with a 'success' indicator, acknowledging receipt. However, after doing so, it also returns a value. So it wasn't as trivial as just wrapping a few lines into a callback. So i opted to leave it untouched for this PR and it can be addressed in a follow-on Jira